### DISCUSSION TITLE
Raw text in grid makes figure a listing

### DISCUSSION THREAD

**Discussion by GamerBene19** [typst/typst#3844] [General]
### Description

When there is raw text in a grid in a figure it becomes a listing. 
@tingerrr (I assume this is his account here) told me to report this [on discord](https://discord.com/channels/1054443721975922748/1088371919725793360/1222595427912781864)

Code to reproduce:
```
#figure(
  grid(
    gutter: auto,
    [I'm a Figure],    
  ),
  caption: "foo"
)

#figure(
  grid(
    gutter: auto,
    [
      ```
      I'm a Listing
      ```
    ],    
  ),
  caption: "bar"
)
```
produces 
![image](https://github.com/typst/typst/assets/20821976/78dd2eff-e26a-4516-a109-955847cbfc5a)
with typst version `typst 0.11.0 (2bf9f95d)`

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by PgBiel** [comment]
It seems to just query for the first element which can be used to represent a figure, which is interesting. I assume this is to avoid interference with block / box / etc. being used to decorate the figure.

https://github.com/typst/typst/blob/105d7156f8f9d95e16b3eefdf0fa97e5be7fbe5b/crates/typst/src/model/figure.rs#L236-L242

Whether or not this is desirable is possible to discuss, but I'd say that one should explicitly specify the kind when the default isn't "good enough" either way (with `kind: image` for the default and `kind: raw` or `kind: table` or `kind: "custom string"` for other options).

**Comment by tingerrr** [comment]
I think for something like grid, it kind of makes sense if all of its items had the same element type, but assuming a kind for `figure(grid(raw(...), image(...)))` seems weird to me.

**Comment by laurmaedje** [comment]
In principle, everything is working as intended here, so I'll convert this into a discussion.
